### PR TITLE
Added Dependency exclusions to remove dependency issues found by enforcer plugin.

### PR DESCRIPTION
**Pull request by agustinlacombadev**
Using enfrocer pluging to remove dependency issues.

[Link](https://github.com/BroadleafCommerce/QA/issues/4583)

**Release notes: Libs upgraded**
- esapi from 2.2.3.0 to 2.2.3.1
- commons-collections4 from 4.1 to 4.2

### CODE REVIEW

File: admin/broadleaf-contentmanagement-module/pom.xml [hunk 1]
```diff
@@ -109,5 +109,16 @@
             <artifactId>spring-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (riteshadhikari17) [inline admin/broadleaf-contentmanagement-module/pom.xml]: This is defined in the top level pom line 689. Can this be removed here and Use top level pom to exclude?

File: admin/broadleaf-open-admin-platform/pom.xml [hunk 2]
```diff
@@ -211,5 +211,33 @@
             <artifactId>spock-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (riteshadhikari17) [inline admin/broadleaf-open-admin-platform/pom.xml]: Same as above. Also lets not define versions in these inner level poms. It is easy to maintain versions if all versions are in the top level pom. Line 688 in Broadleaf/pom.

File: admin/broadleaf-open-admin-platform/pom.xml [hunk 3]
```diff
@@ -211,5 +211,33 @@
...
+            <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.ant</groupId>
+                    <artifactId>ant</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>commons-fileupload</groupId>
```
Comment (riteshadhikari17) [inline admin/broadleaf-open-admin-platform/pom.xml]: Same as above. This is defined in line 674. Add exclusions there.

File: common/pom.xml [hunk 4]
```diff
@@ -427,5 +411,250 @@
             <groupId>com.googlecode.htmlcompressor</groupId>
             <artifactId>htmlcompressor</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.javascript</groupId>
+            <artifactId>closure-compiler-unshaded</artifactId>
```
Comment (riteshadhikari17) [inline common/pom.xml]: Same as above. This is defined in line 1313. Add exclusions there.

File: common/pom.xml [hunk 5]
```diff
@@ -427,5 +411,250 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>javax.servlet</groupId>
+                    <artifactId>servlet-api</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+
+        <dependency>
```
Comment (riteshadhikari17) [inline common/pom.xml]: Line 674

File: common/pom.xml [hunk 6]
```diff
@@ -427,5 +411,250 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>javax.servlet</groupId>
+                    <artifactId>servlet-api</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>commons-beanutils</groupId>
```
Comment (riteshadhikari17) [inline common/pom.xml]: Line 663

File: common/pom.xml [hunk 7]
```diff
@@ -427,5 +411,250 @@
...
+                    <groupId>javax.el</groupId>
+                    <artifactId>javax.el-api</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>org.owasp.esapi</groupId>
+            <artifactId>esapi</artifactId>
+            <version>2.2.3.0</version>
```
Comment (riteshadhikari17) [inline common/pom.xml]: line 1252

File: core/broadleaf-framework-web/pom.xml [hunk 8]
```diff
@@ -112,5 +108,17 @@
             <groupId>org.easymock</groupId>
             <artifactId>easymockclassextension</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.social</groupId>
+            <artifactId>spring-social-web</artifactId>
```
Comment (riteshadhikari17) [inline core/broadleaf-framework-web/pom.xml]: Line 1038

File: core/broadleaf-framework/pom.xml [hunk 9]
```diff
@@ -157,5 +157,20 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-test</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.solr</groupId>
```
Comment (riteshadhikari17) [inline core/broadleaf-framework/pom.xml]: Line 878

File: core/broadleaf-profile/pom.xml [hunk 10]
```diff
@@ -80,5 +80,17 @@
             <groupId>org.easymock</groupId>
             <artifactId>easymockclassextension</artifactId>
         </dependency>
+        <dependency>
```
Comment (riteshadhikari17) [inline core/broadleaf-profile/pom.xml]: Line 633

**Review by riteshadhikari17 — DISMISSED** [review]
In general lets use Top level pom to exclude and lets not use version and scope in the inner poms

File: common/pom.xml (hunk 4 above)
Comment (agustinlacombadev) [reply common/pom.xml]: If I move this to master pom, enforcer fails . the other changes were added correctly

Comment (agustinlacombadev) [reply common/pom.xml]: originally we had that dependency in commons pom, If I remove it completelly, the build fails as well.

Comment (riteshadhikari17) [reply common/pom.xml]: Ok but we should remove version and scope. You could have added exclusions in the dependency that was defined above already.

Comment (agustinlacombadev) [reply common/pom.xml]: done!

**Review by oleksiimiroshnyk — APPROVED** [review]

**Pull request merged**